### PR TITLE
Update code-splitting.md

### PR DESCRIPTION
**Pull request by nicky-lau**
add dynamic import plugin

### CODE REVIEW

**Comment by montogeek** [comment]
Hi @nicky-lau, is that plugin necessary?

**Comment by montogeek** [comment]
Also, this would likely change for webpack 4 since `CommonsChunkPlugin` was removed entirely

**Comment by nicky-lau** [comment]
@montogeek yes,this plugin is necessary, if not using it, the following error occurs at webpack 3.11.0:
```
Module build failed: SyntaxError: 'import' and 'export' may only appear at the top level
```

also, `dynamic import` is just for SPA, but `CommonChunkPlugin` is working for mulit entry app,

**Comment by skipjack** [comment]
@nicky-lau have you specified `modules: false` in your `.babelrc` to allow webpack to process ES6 module syntax? I use dynamic `import()`s a lot and haven't needed that plugin.

**Comment by nicky-lau** [comment]
@skipjack thank ur answer.
here is my `.babelrc`:
```
{
	"presets": [
		["env", {
			"targets": {
				"browsers": ["last 2 versions", "> 1%"]
			},
			"modules": false
		}]
	]
}
```
entry file: 
``` javascript
const arr = [1, 6, 3, 4, 5, 2];

if (arr.includes(2)) {
	import(/* webpackChunkName: "lodash" */'lodash')
	.then(_ => {
		console.log(_.join(['hello', 'world', '!'], ' '));
	})
	// require.ensure([], () => {
	// 	const _ = require('lodash');
	// 	console.log(_.join(['Hello', 'World', '!'], ' '));
	// }, 'lodash')
}
```
error info: 
``` 
Module build failed: SyntaxError: 'import' and 'export' may only appear at the top level
```

**Comment by skipjack** [comment]
@nicky-lau could you create a demo repo? @montogeek can you take a look at that with @nicky-lau can see if it's something else that needs fixing and if not we can continue this PR to get it merged? Embarrassed to say most of my projects still use the `es2015` and `stage-*` presets so maybe it has something to do with `babel-preset-env` (which I know is now the recommended practice).

**Comment by montogeek** [comment]
It would say that the plugin is necessary to allow Babel to parse those `import`s.
@skipjack `babel-preset-env` does not really add plugins automatically, it only add polyfills and transforms.

File: src/content/guides/code-splitting.md [hunk 1]
```diff
@@ -126,6 +131,20 @@ __webpack.config.js__
       index: './src/index.js',
       another: './src/another-module.js'
     },
++  module: {
++     rules: [
++       {
++         test: /\.js$/,
++         exclude: /node_modules/,
++         ues: {
```
Comment (dhruvdutt) [inline src/content/guides/code-splitting.md]: Type ~`ues`~ => `uses`

File: src/content/guides/code-splitting.md [hunk 2]
```diff
@@ -126,6 +131,20 @@ __webpack.config.js__
...
++           loader: 'babel-loader',
```
Comment (dhruvdutt) [inline src/content/guides/code-splitting.md]: Can you please keep consistency in object keys quotes? Some have double, some have single.

File: src/content/guides/code-splitting.md [hunk 3]
```diff
@@ -126,6 +131,20 @@ __webpack.config.js__
...
++           "plugins": [
++           "syntax-dynamic-import" // important!
```
Comment (dhruvdutt) [inline src/content/guides/code-splitting.md]: Wrong indentation.

File: src/content/guides/code-splitting.md (hunk 2 above)
Comment (nicky-lau) [reply src/content/guides/code-splitting.md]: :+1:

File: src/content/guides/code-splitting.md [hunk 4]
```diff
@@ -126,6 +131,18 @@ __webpack.config.js__
       index: './src/index.js',
       another: './src/another-module.js'
     },
++  module: {
```
Comment (dhruvdutt) [inline src/content/guides/code-splitting.md]: Wrong indentation.

Comment (nicky-lau) [reply src/content/guides/code-splitting.md]: thank you for pointing that.

**Review by dhruvdutt — DISMISSED** [review]
LGTM. Please rebase.

File: src/content/guides/code-splitting.md [hunk 5]
```diff
@@ -114,6 +114,11 @@ The first of these two points is definitely an issue for our example, as `lodash
 
 The [`CommonsChunkPlugin`](/plugins/commons-chunk-plugin) allows us to extract common dependencies into an existing entry chunk or an entirely new chunk. Let's use this to de-duplicate the `lodash` dependency from the previous example:
 
+First, let's install this plugin: `syntax-dynamic-import`
+``` npm
+npm install --save-dev babel-plugin-syntax-dynamic-import
+```
```
Comment (jeremenichelli) [inline src/content/guides/code-splitting.md]: Hi @nicky-lau thanks for the contribution. IT's important to note users that this is only needed if they are transpiling their code with babel which might not be the case.

So I suggest changing this to a `W>` blockquote to inform users that **if** they are using Babel, they will need to include this so the Babel understands the dynamic import syntax, and remove it from the code below. Would you be able to do this?

**Pull request closed without merging**